src/utils/index.js

```diff
@@ -74,3 +74,36 @@ export function getGitValues(url) {
 
   return { gitServer, gitOrg, gitRepo: `${gitRepo}.git` };
 }
+
+// Update the status of steps that follow a step with an error
+export function updateUnexecutedSteps(steps) {
+  if (typeof steps === 'undefined') {
+    return steps;
+  }
+  let errorIndex = steps.length - 1;
+  return steps.map((step, index) => {
```

Comment:
I think it's worth raising https://github.com/tektoncd/pipeline/issues/1439 in the next working group meeting at least to ask for feedback on the issue.

It also looks like the behaviour when cancelling a PipelineRun is different, I see steps marked as Waiting instead of Completed.

Having this workaround in the dashboard for now resolves the immediate user-facing issue for users of the dashboard, but users of kubectl or other tools still have the same misleading information presented (i.e. a step that hasn't executed is marked as Completed), and each tool needs to handle this themselves.